include/roaring/portability.h

```diff
@@ -328,6 +328,7 @@ static inline int hamming(uint64_t x) {
 #endif
 
 #define CROARING_TARGET_AVX2 CROARING_TARGET_REGION("avx2,bmi,pclmul,lzcnt")
+#define CROARING_TARGET_AVX512 CROARING_TARGET_REGION("bmi2,avx512f,avx512bw,avx512dq")
```

Comment:
It looks to me like you only check for the AVX-512 bit and then assume that avx512bw and avx512dq follow, but that's not true on Xeon Phi accelerators (and I own one of those, they do exist).

Furthermore, you start the comment with a reference to icelake. 

So I recommend going all out and checking for everything up to VBMI2...

https://github.com/simdjson/simdjson/blob/005054916d2d5b89c1a2f7916704444b054e820c/include/simdjson/internal/isadetection.h#L70-L78

Then you can target the whole range...

"avx512f,avx512dq,avx512cd,avx512bw,avx512vbmi,avx512vbmi2,avx512vl,avx2,bmi,pclmul,lzcnt"

This opens up crazily fast bitmap decoding...

https://lemire.me/blog/2022/05/10/faster-bitset-decoding-using-intel-avx-512/

You don't need to implement that in this PR, but targeting Zen 4 and Ice Lake right away, allows us to add this easily in the future.

Furthermore, Ice Lake and Zen 4 do not suffer from frequency throttling, like the previous AVX-512-capable chips. This makes it much more interesting to use AVX-512 *just* on recent chips capable of VBMI2.

Reply:
i will add checking